### PR TITLE
Deep tree handler customization support

### PR DESCRIPTION
**Pull request by idlesign**

### CODE REVIEW

File: docs/source/customization.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+It might be that you need some special tree items ordering in a menu, or you want to render
+in a huge site tree with all articles titles that are described by one tree item in Django admin,
+or god knows what else.
+
+*django-sitetree* can facilitate on that as it allows tree handler customization
+with the help of `SITETREE_CLS` setting.
+
+1. Subclass `sitetreeapp.SiteTree` and place that class into a separate module for convenience.
+2. Override methods you need for customization (usually `.apply_hook()`).
```
Comment (lorenz-bienek) [inline docs/source/customization.rst]: From that description I don't really understand how .apply_hook() comes into play.

Comment (idlesign) [reply docs/source/customization.rst]: `.appy_hook()` is most generic — this description doesn't necessarily covers your very case from #263 
Rather you override `.get_permissions()`.

Comment (lorenz-bienek) [reply docs/source/customization.rst]: Yeah, thats right. I just wanted to point out, that somebody who reads this sentence might not understand if he has to overwrite .apply_hook() or if this is just one option to deal with a usecase.

But I guess if people are doing such things they must have a look at the code and then everything is pretty self-explanatory anyways.

Comment (idlesign) [reply docs/source/customization.rst]: I see. If you can propose a better wording, I'd be glad.

Comment (lorenz-bienek) [reply docs/source/customization.rst]: I'd be a little bit more explicit, like this:

You can now overwrite `.apply_hook()` to manipulate values at pre-defined hooks. You can also overwrite any other methods to customize to your exact needs.

Comment (idlesign) [reply docs/source/customization.rst]: Done. Thank you.

**Pull request merged**